sim_pipeline/sim_lsst_galaxy_galaxy_lens_simulation.py

```diff
@@ -0,0 +1,283 @@
+
+kwargs_model = {'lens_model_list': ['SIE', 'SHEAR'],  # list of lens models to be used
+                'lens_light_model_list': ['SERSIC_ELLIPSE'],  # list of unlensed light models to be used
+                'source_light_model_list': ['SERSIC_ELLIPSE'],  # list of extended source models to be used
+                }
+# here we define the numerical options used in the ImSim module.
+# Have a look at the ImageNumerics class for detailed descriptions.
+# If not further specified, the default settings are used.
+kwargs_numerics = {'point_source_supersampling_factor': 1, 'supersampling_factor': 3}
+
+g_band = ObservationLSST('g').get_kwargs()
+r_band = ObservationLSST('r').get_kwargs()
+i_band = ObservationLSST('i').get_kwargs()
+
+sim_g = SimAPI(numpix=numpix, kwargs_single_band=g_band, kwargs_model=kwargs_model)
+sim_r = SimAPI(numpix=numpix, kwargs_single_band=r_band, kwargs_model=kwargs_model)
+sim_i = SimAPI(numpix=numpix, kwargs_single_band=i_band, kwargs_model=kwargs_model)
+
+imSim_g = sim_g.image_model_class(kwargs_numerics)
+imSim_r = sim_r.image_model_class(kwargs_numerics)
+imSim_i = sim_i.image_model_class(kwargs_numerics)
+
+path = os.getcwd()
+dirpath, _ = os.path.split(path)
+module_path, _ = os.path.split(dirpath)
+skypy_config = os.path.join(module_path, 'data\SkyPy\lsst-like.yml') #read the file
+skycon = Skypipeline(skypy_config).run_pipeline()# access the results
+print(skycon.state)
+
+class LenstronomyConfig:
+    """
+    creat a class for getting galaxy parameters for g, r and i bands, distance, redshift
+    """
+    def __init__(self, pipeline, mag_cut, z_min, z_max, galaxy_type, position_scatter):
+        """
+        Constructor of the class that initializes the object with specified parameters
+
+        :param pipeline: the results access from Skypipeline
+        :param mag_cut
+        :param z_min: minimum for the redshift in the cut
+        :param z_max: maximum for the redshift in the cur
+        :param galaxy_type: 'blue' or 'red' (source or lens)
+        :param position_scatter:
+        """
+        self.pipeline = pipeline
+        self.mag_cut = mag_cut
+        self.z_min = z_min
+        self.z_max = z_max
+        self.galaxy_type = galaxy_type
+        self.position_scatter = position_scatter
+        self.cosmo = cosmo
+
+    def get_source(self):
+        """
+        Define the source cut based on redshift, magnitude and galaxy type
+        return the information
+        """
+        source_cut = (self.pipeline[self.galaxy_type]['z'] > self.z_min) & (
+                    self.pipeline[self.galaxy_type]['z'] < self.z_max) & (
+                                 self.pipeline[self.galaxy_type]['mag_g'] < self.mag_cut)
+        pipeline_cut = self.pipeline[self.galaxy_type][source_cut]
+        n = len(pipeline_cut)
+        index = random.randint(0, n - 1)
+        source = pipeline_cut[index]
+        return source
+
+    def get_params(self):
```

Comment:
this function should have a more specific name indicating what is done here

Reply:
I would also split this function into 
1. where  some of the parameters are drawn with some randomness and
2. how these parameters are turned into lenstronomy compatible functions